tests/features/steps/artifacts-poetry-plugin-steps.py

```diff
@@ -15,52 +13,107 @@
         )
 
 
-@given("a Python project with dependencies on artifacts {keys}")
+@given("a Python project with dependencies on package {keys}")
 def step_impl(context, keys):
-    shutil.copy(TEMPLATE_PY_PROJECT, PY_PROJECT)
-    with open(PY_PROJECT, "a") as py_project:
-        py_project.write("\n[tool.poetry.dependencies]\n")
+    with open(POETRY_LOCK, "w") as poetry_lock:
+        poetry_lock.write("\n[metadata]\n")
+        poetry_lock.write('lock-version = "2.0"\n')
+        poetry_lock.write('python-versions = "^3.8"\n\n')
         for key in keys.split(","):
             package = context.packages[key]
-            py_project.write(f'{package.name} = "{package.pretty_version}"\n')
-        py_project.write(f'python = "^3.8"')
-    subprocess.run(["poetry", "install", "--no-root"], cwd=TEST_RESOURCES_DIR)
+            files = package.files
+            poetry_lock.write("[[package]]\n")
+            poetry_lock.write(f'name = "{package.name}"\n')
+            poetry_lock.write(f'version = "{package.pretty_version}"\n')
+            poetry_lock.write(f'description = "Test"\n')
+            poetry_lock.write(f"optional = false\n")
+            poetry_lock.write(f'python-versions = ">=2.6, !=3.0.*, !=3.1.*, !=3.2.*"\n')
+            poetry_lock.write(f"files = [\n")
+            for file in files:
+                poetry_lock.write(
+                    "{"
+                    + f'file = "{file}", hash = "sha256:ebda1a6c9e5bfe95c5f9f0a2794e01c7098b3dde86c10a95d8621c5907ff6f1c"'
+                    + "},\n"
+                )
+            poetry_lock.write("]\n\n")
 
 
-@when("the Python dependency search is triggered")
+@when("the Python deployable dependency is triggered.")
 def step_impl(context):
     poetry = Factory().create_poetry(Path(POETRY_LOCK))
-    context.response_packages = getPackages(poetry)
+    deploy_packages, non_deploy_packages = get_deploy_packages(
+        poetry, TEST_RESOURCES_DIR
+    )
+    context.deploy_packages = deploy_packages
+    context.non_deploy_packages = non_deploy_packages
+
 
+def get_package_set(packages):
+    package_set = set()
+    for package in packages:
+        package_set.add((package.name, package.pretty_version))
+    return package_set
 
-@then("artifacts {keys} are located in the local cache")
+
+def get_package_set_from_keys(keys, packages):
+    package_set = set()
+    for key in keys.split(","):
+        if key in packages:
+            package = packages[key]
+            package_set.add((package.name, package.pretty_version))
+    return package_set
+
+
+@then("package {keys} are able to be deployed to an alternate repository.")
+def step_impl(context, keys):
+    deploy_package_set = get_package_set(context.deploy_packages)
+    compare_package_set = get_package_set_from_keys(keys, context.packages)
+    assert compare_package_set.issubset(deploy_package_set)
+
+
+@given("no wheel file associated with package {keys}")
```

Comment:
A: Glue code tends to be more navigable if you group the givens together, the whens together, and the thens together.  Because clauses tend to be reused, there's usually no obvious linear structure outside of just grouping by when in the scenario the clause is listed.